fdbserver/storageserver.actor.cpp

```diff
@@ -287,7 +287,7 @@ class ShardInfo : public ReferenceCounted<ShardInfo>, NonCopyable {
 	}
 
 	StorageServerShard toStorageServerShard() const {
-		StorageServerShard::ShardState st;
+		StorageServerShard::ShardState st = StorageServerShard::NotAssigned;
```

Comment:
Another option here is to remove the if statement at line 297. We don't need it because it will always be true if you get passed the assert.

Reply:
Removed the if statement, kept the initialization as well.